discovery-infra/tests/base_test.py

```diff
@@ -34,33 +36,34 @@ def get_nodes(self) -> Callable:
         nodes_data = dict()
 
         @JunitTestCase()
-        def get_nodes_func(config: TerraformConfig = TerraformConfig()):
+        def get_nodes_func(config: Optional[TerraformConfig] = None):
+            if not config:
+                config = TerraformConfig()
+
             if "nodes" in nodes_data:
                 return nodes_data["nodes"]
 
-            net_asset = NetworkAssets()
-            config.net_asset = net_asset.get()
+            with file_lock_context(consts.NODES_DEPLOYMENT_LOCK_FILE):
```

Comment:
filelock should be part of asset and not base test as it will allow to use asset in other places if needsed

Reply:
there should be also a file lock in the base test since the parallel test will fail in case two processes allocate the same resources at the same time.

Reply:
it should be handled by lock inside assets file.

Reply:
but then we are missing the whole point of this PR.

Reply:
Two assets from two processes can allocate the same resources even with the internal lock.
It looks on the current free resources and until the nodes are not up the same resources will be free.

Reply:
I will change it to write the allocated resources to a shared file.